src/core/RemotePlugin.cpp

```diff
@@ -481,37 +470,10 @@ void RemotePlugin::hideUI()
 
 void RemotePlugin::resizeSharedProcessingMemory()
 {
-	const size_t s = ( m_inputCount+m_outputCount ) * Engine::audioEngine()->framesPerPeriod() * sizeof( float );
-	if( m_shm != nullptr )
-	{
-#ifdef USE_QT_SHMEM
-		m_shmObj.detach();
-#else
-		shmdt( m_shm );
-		shmctl( m_shmID, IPC_RMID, nullptr );
-#endif
-	}
-
-	static int shm_key = 0;
-#ifdef USE_QT_SHMEM
-	do
-	{
-		m_shmObj.setKey( QString( "%1" ).arg( ++shm_key ) );
-		m_shmObj.create( s );
-	} while( m_shmObj.error() != QSharedMemory::NoError );
-
-	m_shm = (float *) m_shmObj.data();
-#else
-	while( ( m_shmID = shmget( ++shm_key, s, IPC_CREAT | IPC_EXCL |
-								0600 ) ) == -1 )
-	{
-	}
-
-	m_shm = (float *) shmat( m_shmID, 0, 0 );
-#endif
-	m_shmSize = s;
-	sendMessage( message( IdChangeSharedMemoryKey ).
-				addInt( shm_key ).addInt( m_shmSize ) );
+	const size_t s = (m_inputCount + m_outputCount) * Engine::audioEngine()->framesPerPeriod();
+	m_audioBuffer.create(QUuid::createUuid().toString().toStdString(), s);
```

Comment:
What if this call throws due to very unlikely failures?

Reply:
The existing code didn't handle this very well (it just retried in a loop, assuming the only error would be key collisions, which shouldn't happen now with UUIDs), so I didn't bother doing it better. I'll change it to at least catch and log any exception thrown here.